lib/db/observed.go

```diff
@@ -111,6 +111,42 @@ func (db *Lowlevel) RemovePendingFolder(id string) {
 	}
 }
 
+func (db *Lowlevel) RemovePendingFoldersBeforeTime(device protocol.DeviceID, oldest time.Time) (uint, error) {
+	prefixKey, err := db.keyer.GeneratePendingFolderKey(nil, device[:], nil)
+	if err != nil {
+		return 0, err
+	}
+	iter, err := db.NewPrefixIterator(prefixKey)
+	if err != nil {
+		return 0, err
+	}
+	defer iter.Release()
+	oldest = oldest.Round(time.Second)
+	var count uint
+	for iter.Next() {
+		var bs []byte
+		var of ObservedFolder
+		if bs, err = db.Get(iter.Key()); err != nil {
+			// Keep inaccessible entries
+			continue
+		} else if err = of.Unmarshal(bs); err != nil {
+			// Keep invalid entries
+			continue
```

Comment:
Why keep all of these dodgy things, and skim over them in silence?

Reply:
Might just as well delete them, just like `PendingFolders()` does. I did not plan on that since it's already done there and I find it confusing if the functions removes anything other than it was requested to do.

So should I remove them here?

Reply:
I guess warn and remove? Definately not sit in silence and do nothing.